## Estadística Aplicada

Actividad 6a - 26 septiembre

Karla Mariana Ley Acero- 2076201

Grupo 42

1. Importa las librerías básicas y especializadas que consideres necesarias

In [58]:
# Librerías básicas

import pandas as pd 
import numpy as np 
import matplotlib as plt
%matplotlib inline

In [59]:
 # Librerías especializadas
from lifelines import CoxPHFitter, ExponentialFitter, WeibullFitter, LogNormalFitter

2. Importar la base de datos

In [60]:
desercion = pd.read_csv('https://raw.githubusercontent.com/jimmyzac/Estadistica-Aplicada-FCFM-UANL/main/bases_datos/desercion_laboral.csv')

3. Depurar la base y estadísticas descriptivas

In [61]:
desercion.head()

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0           3.8         5.3          2                 157         3   
1           8.0         8.6          5                 262         6   
2           1.1         8.8          7                 272         4   
3           7.2         8.7          5                 223         5   
4           3.7         5.2          2                 159         3   

   accidente  renunciar  ascenso_5years departamento salario  
0          0          1               0       ventas    bajo  
1          0          1               0       ventas   medio  
2          0          1               0       ventas   medio  
3          0          1               0       ventas    bajo  
4          0          1               0       ventas    bajo

In [62]:
# Identificar los missing values 
desercion.isnull().sum()

satisfaccion           0
evaluacion             0
proyectos              0
promedio_horas_mes     0
semestre               0
accidente              0
renunciar              0
ascenso_5years         0
departamento          26
salario                0
dtype: int64

En la variable despartamentos hay 26 missing values 

In [63]:
# Eliminar los missing values
desercion = desercion.dropna() 

In [64]:
desercion

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0               3.8         5.3          2                 157         3   
1               8.0         8.6          5                 262         6   
2               1.1         8.8          7                 272         4   
3               7.2         8.7          5                 223         5   
4               3.7         5.2          2                 159         3   
...             ...         ...        ...                 ...       ...   
14994           4.0         5.7          2                 151         3   
14995           3.7         4.8          2                 160         3   
14996           3.7         5.3          2                 143         3   
14997           1.1         9.6          6                 280         4   
14998           3.7         5.2          2                 158         3   

       accidente  renunciar  ascenso_5years departamento salario  
0              0          1               0       ventas    bajo  
1              0          1               0       ventas   medio  
2              0          1               0       ventas   medio  
3              0          1               0       ventas    bajo  
4              0          1               0       ventas    bajo  
...          ...        ...             ...          ...     ...  
14994          0          1               0      soporte    bajo  
14995          0          1               0      soporte    bajo  
14996          0          1               0      soporte    bajo  
14997          0          1               0      soporte    bajo  
14998          0          1               0      soporte    bajo  

[14973 rows x 10 columns]

#### b) Verificar que no haya valores duplicados con dataframe.duplicated().sum() en caso de haber valores duplicados eliminarlos con dataframe=dataframe.drop_duplicates(keep='first').reset_index(drop=True)

In [65]:
desercion.duplicated().sum()

2982

Existen 2982 valores duplicados

In [66]:
# Eliminar los 2982 valores duplicados 

desercion=desercion.drop_duplicates(keep='first').reset_index(drop=True)
desercion

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0               3.8         5.3          2                 157         3   
1               8.0         8.6          5                 262         6   
2               1.1         8.8          7                 272         4   
3               7.2         8.7          5                 223         5   
4               3.7         5.2          2                 159         3   
...             ...         ...        ...                 ...       ...   
11986           3.7         4.7          2                 152         3   
11987           8.4         9.9          4                 267         5   
11988           4.1         4.6          2                 151         3   
11989           7.6         9.2          4                 239         5   
11990           1.1         8.7          6                 306         4   

       accidente  renunciar  ascenso_5years            departamento salario  
0              0          1               0                  ventas    bajo  
1              0          1               0                  ventas   medio  
2              0          1               0                  ventas   medio  
3              0          1               0                  ventas    bajo  
4              0          1               0                  ventas    bajo  
...          ...        ...             ...                     ...     ...  
11986          0          1               0  desarrollo de producto    bajo  
11987          0          1               0  desarrollo de producto    bajo  
11988          0          1               0  desarrollo de producto    bajo  
11989          0          1               0  desarrollo de producto    bajo  
11990          0          1               0             informática    bajo  

[11991 rows x 10 columns]

#### c) Verificar que todas las variables sean numéricas, si no son numéricas hacerlas numéricas, en el caso de las categóricas convertirlas en dummies y conservar n-1 dummies

In [67]:
# Checar que tipo de variables son 

desercion.dtypes

satisfaccion          float64
evaluacion            float64
proyectos               int64
promedio_horas_mes      int64
semestre                int64
accidente               int64
renunciar               int64
ascenso_5years          int64
departamento           object
salario                object
dtype: object

Departamento y salario son variables categóricas por lo que hay que convertirlas en dummies

In [68]:
# Checar cuales son los valores de las variables categóricas para seleccionar cual eliminar

desercion['departamento'].value_counts()

departamento
ventas                    3239
mantenimiento             2244
soporte                   1821
informática                976
Investigación              694
desarrollo de producto     686
mercadotecnia              673
contabilidad               621
recursos humanos           601
administración             436
Name: count, dtype: int64

In [69]:
desercion['salario'].value_counts()

salario
bajo     5740
medio    5261
alto      990
Name: count, dtype: int64

In [70]:
# Convertir la variable salario a dummy 

salarios_dummy = desercion['salario']
#Hacer dummies
salarios_dummy= pd.get_dummies(salarios_dummy, prefix='salario', drop_first = True).astype(int)
salarios_dummy.head()

salario_bajo  salario_medio
0             1              0
1             0              1
2             0              1
3             1              0
4             1              0

In [71]:
# Convertir variable departamento

departamentos_dummy = desercion['departamento']
departamentos_dummy = pd.get_dummies(departamentos_dummy, prefix='departamentos', drop_first = True).astype(int)
departamentos_dummy.head()

departamentos_administración  departamentos_contabilidad  \
0                             0                           0   
1                             0                           0   
2                             0                           0   
3                             0                           0   
4                             0                           0   

   departamentos_desarrollo de producto  departamentos_informática  \
0                                     0                          0   
1                                     0                          0   
2                                     0                          0   
3                                     0                          0   
4                                     0                          0   

   departamentos_mantenimiento  departamentos_mercadotecnia  \
0                            0                            0   
1                            0                            0   
2                            0                            0   
3                            0                            0   
4                            0                            0   

   departamentos_recursos humanos  departamentos_soporte  departamentos_ventas  
0                               0                      0                     1  
1                               0                      0                     1  
2                               0                      0                     1  
3                               0                      0                     1  
4                               0                      0                     1

In [72]:
# Eliminar las columnas originales de salario y departamento DataFrame y agregar las nuevas

desercion = desercion.drop(['salario', 'departamento'], axis=1)


In [73]:
desercion = pd.concat([desercion, departamentos_dummy], axis=1)
desercion = pd.concat([desercion, salarios_dummy], axis=1)

d) Calcular estadísticas descriptivas

In [74]:
desercion.describe()

satisfaccion    evaluacion     proyectos  promedio_horas_mes  \
count  11991.000000  11991.000000  11991.000000        11991.000000   
mean       6.296581      7.166825      3.802852          200.473522   
std        2.410700      1.683426      1.163238           48.727813   
min        0.900000      3.600000      2.000000           96.000000   
25%        4.800000      5.700000      3.000000          157.000000   
50%        6.600000      7.200000      4.000000          200.000000   
75%        8.200000      8.600000      5.000000          243.000000   
max       10.000000     10.000000      7.000000          310.000000   

           semestre     accidente     renunciar  ascenso_5years  \
count  11991.000000  11991.000000  11991.000000    11991.000000   
mean       3.364857      0.154282      0.166041        0.016929   
std        1.330240      0.361234      0.372133        0.129012   
min        2.000000      0.000000      0.000000        0.000000   
25%        3.000000      0.000000      0.000000        0.000000   
50%        3.000000      0.000000      0.000000        0.000000   
75%        4.000000      0.000000      0.000000        0.000000   
max       10.000000      1.000000      1.000000        1.000000   

       departamentos_administración  departamentos_contabilidad  \
count                  11991.000000                11991.000000   
mean                       0.036361                    0.051789   
std                        0.187194                    0.221610   
min                        0.000000                    0.000000   
25%                        0.000000                    0.000000   
50%                        0.000000                    0.000000   
75%                        0.000000                    0.000000   
max                        1.000000                    1.000000   

       departamentos_desarrollo de producto  departamentos_informática  \
count                          11991.000000               11991.000000   
mean                               0.057210                   0.081394   
std                                0.232252                   0.273451   
min                                0.000000                   0.000000   
25%                                0.000000                   0.000000   
50%                                0.000000                   0.000000   
75%                                0.000000                   0.000000   
max                                1.000000                   1.000000   

       departamentos_mantenimiento  departamentos_mercadotecnia  \
count                  11991.00000                 11991.000000   
mean                       0.18714                     0.056125   
std                        0.39004                     0.230173   
min                        0.00000                     0.000000   
25%                        0.00000                     0.000000   
50%                        0.00000                     0.000000   
75%                        0.00000                     0.000000   
max                        1.00000                     1.000000   

       departamentos_recursos humanos  departamentos_soporte  \
count                    11991.000000           11991.000000   
mean                         0.050121               0.151864   
std                          0.218204               0.358904   
min                          0.000000               0.000000   
25%                          0.000000               0.000000   
50%                          0.000000               0.000000   
75%                          0.000000               0.000000   
max                          1.000000               1.000000   

       departamentos_ventas  salario_bajo  salario_medio  
count          11991.000000  11991.000000   11991.000000  
mean               0.270119      0.478692       0.438746  
std                0.444040      0.499567       0.496254  
min                0.000000      0.000000       0.000000  
25%             

4. Estimar el modelo de COX e interpretar cada uno de los coeficientes

In [75]:
modelo_cox = CoxPHFitter().fit(desercion, 'semestre', 'renunciar')

In [76]:
modelo_cox.print_summary()

<lifelines.CoxPHFitter: fitted with 11991 total observations, 10000 right-censored observations>
             duration col = 'semestre'
                event col = 'renunciar'
      baseline estimation = breslow
   number of observations = 11991
number of events observed = 1991
   partial log-likelihood = -15884.05
         time fit was run = 2023-09-27 04:12:12 UTC

---
                                       coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%
covariate                                                                                                                                    
satisfaccion                          -0.24       0.78       0.01            -0.26            -0.22                 0.77                 0.80
evaluacion                            -0.01       0.99       0.01            -0.04             0.02                 0.96                 1.02
proyectos                             -0.29       0.75       0.02            -0.33            -0.25                 0.72                 0.78
promedio_horas_mes                     0.00       1.00       0.00             0.00             0.00                 1.00                 1.00
accidente                             -1.25       0.29       0.10            -1.44            -1.05                 0.24                 0.35
ascenso_5years                        -1.38       0.25       0.36            -2.08            -0.68                 0.13                 0.51
departamentos_administración           0.22       1.25       0.18            -0.13             0.57                 0.88                 1.76
departamentos_contabilidad             0.24       1.27       0.14            -0.04             0.52                 0.96                 1.69
departamentos_desarrollo de producto   0.36       1.43       0.14             0.08             0.64                 1.08                 1.90
departamentos_informática              0.29       1.34       0.13             0.03             0.56                 1.03                 1.74
departamentos_mantenimiento            0.43       1.54       0.12             0.20             0.67                 1.22                 1.95
departamentos_mercadotecnia            0.34       1.40       0.14             0.06             0.62                 1.06                 1.86
departamentos_recursos humanos         0.43       1.54       0.14             0.15             0.72                 1.16                 2.05
departamentos_soporte                  0.45       1.57       0.12             0.21             0.69                 1.23                 2.00
departamentos_ventas                   0.35       1.42       0.12             0.12             0.58                 1.13                 1.78
salario_bajo                           1.54       4.66       0.15             1.25             1.83                 3.48                 6.23
salario_medio                          1.15       3.15       0.15             0.85             1.44                 2.35                 4.22

                                       cmp to      z      p   -log2(p)
covariate                                                             
satisfaccion                             0.00 -26.89 <0.005     526.58
evaluacion                               0.00  -0.76   0.44       1.17
proyectos                                0.00 -14.75 <0.005     161.15
promedio_horas_mes                       0.00   3.88 <0.005      13.25
accidente                                0.00 -12.41 <0.005     115.04
ascenso_5years                           0.00  -3.88 <0.005      13.21
departamentos_administración             0.00   1.24   0.21       2.22
departamentos_contabilidad               0.00   1.66   0.10       3.36
departamentos_desarrollo de producto     0.00   2.49   0.01       6.29
departamentos_informática                0.00   2.17   0.03       5.06
departamentos_mantenimiento              0.00   3.60 <0.005      11.64
depa

In [77]:
# Coeficientes sin redondeo
modelo_cox.params_

covariate
satisfaccion                           -0.242337
evaluacion                             -0.010687
proyectos                              -0.291606
promedio_horas_mes                      0.001839
accidente                              -1.246686
ascenso_5years                         -1.380084
departamentos_administración            0.220285
departamentos_contabilidad              0.240389
departamentos_desarrollo de producto    0.360015
departamentos_informática               0.292078
departamentos_mantenimiento             0.431970
departamentos_mercadotecnia             0.338849
departamentos_recursos humanos          0.434338
departamentos_soporte                   0.451085
departamentos_ventas                    0.350547
salario_bajo                            1.538901
salario_medio                           1.147612
Name: coef, dtype: float64

In [78]:
# Exp(coef) sin redondear
modelo_cox.hazard_ratios_

covariate
satisfaccion                            0.784792
evaluacion                              0.989370
proyectos                               0.747063
promedio_horas_mes                      1.001841
accidente                               0.287456
ascenso_5years                          0.251557
departamentos_administración            1.246432
departamentos_contabilidad              1.271743
departamentos_desarrollo de producto    1.433351
departamentos_informática               1.339208
departamentos_mantenimiento             1.540290
departamentos_mercadotecnia             1.403331
departamentos_recursos humanos          1.543940
departamentos_soporte                   1.570015
departamentos_ventas                    1.419844
salario_bajo                            4.659468
salario_medio                           3.150662
Name: exp(coef), dtype: float64

### **Interpretación de los coeficientes**

Tomando en cuenta que:

Rechazamos $H_0$ si p-value < $alpha$

Además:

Cuando coef(+) , entonces exp(coef) > 1 ,es decir , la variable influye de manera positiva (aumenta) ; (exp(coef)-1)
 
Cuando coef(-) , entonces exp(coef)< 1 ,es decir , la variable influye de manera negativa (disminuye) ; (1-exp(coef))




**satisfacción**

0.005 < 0.05 

Rechazamos Ho.

En una escala del uno al 10 por cada punto de satisfacción del empleado con su trabajo el riesgo de que renuncie se reduce un 22% en respecto al punto de satisfacción anterior


**evaluacion**

0.44 < 0.05

No rechazamos Ho. 

La evaluación del empleado respecto a su trabajo no tiene efecto significativo en el riesgo de que este renuncie 

**proyectos**

0.005 < 0.05

Rechazamos Ho. 

Por cada proyecto asignado al empleado el riesgo de que este mismo renuncie disminuye un 25%.

**promedio_horas_mes**

P-value < alfa

0.005 < 0.05

Rechazamos Ho. 

Por cada hora que un empleado trabaje al mes el riesgo de que este renuncie aumenta en un 0.1841%.

**accidente**

0.005 < 0.05

Rechazamos Ho. 

Si el empleado ha sufrido un accidente trabajando el riesgo de que renuncie se reduce en un 71% con respecto a si no hubiera sufrido uno.

**ascenso_5years**

P-value < alfa

0.005 < 0.05

Rechazamos Ho. 

El que un empleado haya sido ascendido durantre los últimos 5 años reduce el riesgo de que renuncie en un 75% respecto a no haber sido ascendido. 

**departamentos__administración**

0.21 < 0.05

No rechazamos Ho. 

El que el empleado trabaje en el departamento de administración no tiene efecto significativo en el riesgo de que renuncie. 

**departamentos__contabilidad**

0.1 < 0.05

No rechazamos Ho. 

El que el empleado trabaje en el departamento de contabilidad no tiene efecto en el riesgo de que renuncie. 

**departamentos__desarrollo de producto**

0.01 < 0.05

Rechazamos Ho. 

El que un empleado trabaje en el departamento de desarrollo de producto aumenta el riesgo de que renuncie en 43% a si trabajara en el departamento de investigación. 

#(variable que comparamos)

**departamentos__informática**

0.03 < 0.05

Rechazamos Ho. 

El que un empleado trabaje en el departamento de informática aumenta el riesgo de que renuncie en 34% a si trabajara en el departamento de investigación. 

**departamentos__mantenimiento**

0.005 < 0.05

Rechazamos Ho. 

El que un empleado trabaje en el departamento de mantenimiento aumenta el riesgo de que renuncie en 54% a si trabajara en el departamento de investigación. 

**departamentos__mercadotecnia**

0.02 < 0.05

Rechazamos Ho.

El que empleado trabaje en el departamento de mercadotecnia aumenta el riesgo de que renuncie en 40% a si trabajara en el departamento de investigación. 

**departamentos__recursos humanos**

0.005 < 0.05 

Rechazamos Ho.

El que un empleado trabaje en el departamento de recursos humanos aumenta el riesgo de que renuncie en 54% a si trabajara en el departamento de investigación. 

**departamentos__soporte**

0.005 < 0.05

Rechazamos Ho. 

El que un empleado trabaje en el departamento de soporte aumenta el riesgo de que renuncie en 57% a si trabajara en el departamento de investigación. 

**departamentos__ventas**

0.005 < 0.05

Rechazamos Ho. 

El que un empleado trabaje en el departamento de ventas aumenta el riesgo de que renuncie en 42% a si trabajara en el departamento de investigación. 

**salarios__bajo**

0.005 < 0.05

Rechazamos Ho. 

El que un empleado reciba un salario bajo aumenta el riesgo de que renuncie en 366% a si recibiera un salario alto. 

**salarios__medio**

0.005 < 0.05

Rechazamos Ho. 

El que un empleado reciba un salario medio aumenta el riesgo de que renuncie en 215% a si recibiera un salario alto. 

### 5 - Con base en la interpretación de los coeficientes,
 ¿qué recomendación podría hacerle a la empresa para que disminuya la cantidad de renuncias de los empleados?

Si un empleado se encuentra satisfecho entonces el riesgo de que este renuncie disminuye y para que este empleado se encuentre satisfecho se ha comprado que debe tener:

Buen salario (considerado alto)

Ser asecendido durante los ultimos 5 años

Con proyectos asignados

Por lo que para las personas que aun no cuentan con los puntos previamente escritos pues hacer lo posible para que los tengan



### 6 - Pruebe si el modelo de Cox cumple el supuesto de riesgos proporcionales, de no hacerlo estime el modelo AFT adecuado

In [80]:
modelo_cox.check_assumptions(desercion , p_value_threshold=0.05)

The ``p_value_threshold`` is set at 0.05. Even under the null hypothesis of no violations, some
covariates will be below the threshold by chance. This is compounded when there are many covariates.
Similarly, when there are lots of observations, even minor deviances from the proportional hazard
assumption will be flagged.

With that in mind, it's best to use a combination of statistical tests and visual tests to determine
the most serious violations. Produce visual plots using ``check_assumptions(..., show_plots=True)``
and looking for non-constant lines. See link [A] below for a full example.



<lifelines.StatisticalResult: proportional_hazard_test>
 null_distribution = chi squared
degrees_of_freedom = 1
             model = <lifelines.CoxPHFitter: fitted with 11991 total observations, 10000 right-censored observations>
         test_name = proportional_hazard_test

---
                                           test_statistic      p  -log2(p)
accidente                            km              0.03   0.87      0.21
                                     rank            1.14   0.29      1.81
ascenso_5years                       km              3.66   0.06      4.17
                                     rank            6.71   0.01      6.71
departamentos_administración         km              5.08   0.02      5.37
                                     rank            5.74   0.02      5.92
departamentos_contabilidad           km              0.09   0.77      0.39
                                     rank            1.14   0.29      1.81
departamentos_desarrollo de producto km              1.03   0.31      1.69
                                     rank            0.85   0.36      1.49
departamentos_informática            km              0.48   0.49      1.03
                                     rank            0.53   0.47      1.10
departamentos_mantenimiento          km              0.47   0.49      1.02
                                     rank            2.26   0.13      2.91
departamentos_mercadotecnia          km              2.66   0.10      3.28
                                     rank            6.33   0.01      6.39
departamentos_recursos humanos       km              0.68   0.41      1.29
                                     rank            2.86   0.09      3.46
departamentos_soporte                km              0.00   0.98      0.03
                                     rank            0.89   0.35      1.53
departamentos_ventas                 km              1.04   0.31      1.70
                                     rank            4.36   0.04      4.76
evaluacion                           km            443.05 <0.005    324.32
                                     rank          427.15 <0.005    312.82
promedio_horas_mes                   km            270.84 <0.005    199.74
                                     rank          307.41 <0.005    226.21
proyectos                            km            574.13 <0.005    419.05
                                     rank          684.93 <0.005    499.11
salario_bajo                         km              0.37   0.54      0.88
                                     rank            0.00   0.96      0.06
salario_medio                        km              0.16   0.69      0.53
                                     rank            0.10   0.75      0.42
satisfaccion                         km           1257.74 <0.005    912.74
                                     rank          845.24 <0.005    614.90



1. Variable 'satisfaccion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'satisfaccion' might be incorrect. That is, there
may be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to specify a functional form.

   Advice 2: try binning the variable 'satisfaccion' using pd.cut, and then specify it in
`strata=['satisfaccion', ...]` in the call in `.fit`. See documentation in link [B] below.

   Advice 3: try adding an interaction term with your time variable. See documentation in link [C]
below.


2. Variable 'evaluacion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'evaluacion' might be incorrect. That is, there may
be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to 

[]

Dos variables no cumplen con el supuesto de riesgos proporcioanales (satisfacción y evaluación). No se puede usar el modelo de COX

**ESTIMACIÓN DE AFT ADECUADO**

In [81]:
# verificar si es Weibull , exponencial o Lognormal. 

modelo_expo = ExponentialFitter().fit(desercion['semestre'],desercion['renunciar'])
modelo_weibull = WeibullFitter().fit(desercion['semestre'],desercion['renunciar'])
modelo_logn = LogNormalFitter().fit(desercion['semestre'],desercion['renunciar'])

In [82]:
## Calcular los AIC de cada modelo

modelo_expo.AIC_, modelo_weibull.AIC_, modelo_logn.AIC_

(15965.45892201228, 12511.512867661068, 11587.92169210356)

In [84]:
pip install lifelines

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [86]:
from lifelines import LogNormalAFTFitter

In [87]:
# Como el modelo de lognormal es el que tiene el menor AIC es el que se va a escoger

modelo_lognAFT = LogNormalAFTFitter().fit(desercion, 'semestre', 'renunciar')
modelo_lognAFT.print_summary()

<lifelines.LogNormalAFTFitter: fitted with 11991 total observations, 10000 right-censored observations>
             duration col = 'semestre'
                event col = 'renunciar'
   number of observations = 11991
number of events observed = 1991
           log-likelihood = -4876.57
         time fit was run = 2023-09-27 04:25:18 UTC

---
                                              coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%
param  covariate                                                                                                                                    
mu_    accidente                              0.26       1.30       0.02             0.22             0.30                 1.25                 1.35
       ascenso_5years                         0.37       1.44       0.06             0.24             0.49                 1.27                 1.63
       departamentos_administración           0.05       1.05       0.04            -0.03             0.13                 0.97                 1.13
       departamentos_contabilidad            -0.04       0.96       0.03            -0.11             0.02                 0.90                 1.02
       departamentos_desarrollo de producto  -0.05       0.95       0.03            -0.12             0.01                 0.89                 1.01
       departamentos_informática             -0.05       0.95       0.03            -0.11             0.01                 0.90                 1.02
       departamentos_mantenimiento           -0.08       0.93       0.03            -0.13            -0.02                 0.88                 0.98
       departamentos_mercadotecnia           -0.04       0.97       0.03            -0.10             0.03                 0.90                 1.03
       departamentos_recursos humanos        -0.09       0.92       0.03            -0.15            -0.02                 0.86                 0.98
       departamentos_soporte                 -0.08       0.92       0.03            -0.14            -0.03                 0.87                 0.97
       departamentos_ventas                  -0.06       0.94       0.03            -0.11            -0.00                 0.90                 1.00
       evaluacion                             0.01       1.01       0.00            -0.00             0.01                 1.00                 1.01
       promedio_horas_mes                    -0.00       1.00       0.00            -0.00            -0.00                 1.00                 1.00
       proyectos                              0.10       1.10       0.01             0.09             0.11                 1.09                 1.11
       salario_bajo                          -0.37       0.69       0.03            -0.43            -0.31                 0.65                 0.73
       salario_medio                         -0.29       0.75       0.03            -0.34            -0.23                 0.71                 0.80
       satisfaccion                           0.07       1.07       0.00             0.06             0.07                 1.06                 1.07
       Intercept                              1.34       3.83       0.04             1.25             1.43                 3.50                 4.18
sigma_ Intercept                             -1.11       0.33       0.02            -1.14            -1.08                 0.32                 0.34

                                              cmp to      z      p   -log2(p)
param  covariate                                                             
mu_    accidente                                0.00  12.98 <0.005     125.59
       ascenso_5years                           0.00   5.80 <0.005      27.19
       departamentos_administración             0.00   1.21   0.23       2.14
       departamentos_contabilidad               0.00  -1.27   0.20       2.30
       departamentos_desarrollo de producto     0.00  -1.5

### 7 - Interprete los coeficientes del modelo AFT

**satisfacción**

0.005 < 0.05 

Rechazamos Ho.

La satisfacción laboral influye de manera importante en el riesgo de que los empleados decidan abandonar su puesto de trabajo. Cada incremento de un punto en la satisfacción laboral de los empleados se traduce en un aumento del 7% en la mediana del tiempo que permanecen en la empresa en comparación con el punto anterior.


**evaluacion**

0.14 > 0.05

No rechazamos Ho. 

La evaluación no tiene un impacto significativo en la decisión de los empleados de dejar su trabajo (renuncia)

**proyectos**

0.005 < 0.05

Rechazamos Ho. 

Por cada número de proyectos asignados al empleado la mediana del tiempo de supervivencia aumenta al 10%


**promedio_horas_mes**

P-value < alfa

0.02 < 0.05

Rechazamos Ho. 

La variable tiene un efecto significativo, cada hora que se trabaja al mes disminuyen la mediana del tiempo de supervivencia 0.0309%.

**accidente**

0.005 < 0.05

Rechazamos Ho. 

La variable muestra un efecto significativo;  el experimentar un accidente dentro de la empresa incrementa el período de supervivencia en un 30% en comparación con aquellos que no han experimentado ningún accidente.

**ascenso_5years**

P-value < alfa

0.005 < 0.05

Rechazamos Ho. 

La variable demuestra tener un efecto significativo, ya que ser ascendido en los últimos cinco años aumenta el 44% la mediana del tiempo de supervivencia en comparación con aquellos que no han experimentado un ascenso.

**departamentos__administración**

0.23 > 0.05

No rechazamos Ho. 

La variable no tiene un efecto significativo, pertenecer al departamento de administración no tiene efecto sobre la mediana del tiempo de supervivencia.

**departamentos__contabilidad**

0.2 < 0.05

No rechazamos Ho. 

La variable no tiene un efecto significativo, pertenecer al departamento de contabilidad no tiene efecto sobre la mediana del tiempo de supervivencia.

**departamentos__desarrollo de producto**

0.12 < 0.05

No rechazamos Ho. 

La variable no tiene un efecto significativo, pertenecer al departamento de desarrollo de producto no tiene efecto sobre la mediana del tiempo de supervivencia.

**departamentos__informática**

0.14 < 0.05

No rechazamos Ho. 

La variable no tiene un efecto significativo, pertenecer al departamento de informática no tiene efecto sobre la mediana del tiempo de supervivencia. 

**departamentos__mantenimiento**

0.01 < 0.05

Rechazamos Ho. 

La variable tiene un efecto significativo, pertenecer al departamento de mantenimiento disminuye la mediana del tiempo de supervivencia 7% con respecto a pertenecer al departamento de investigación.


**departamentos__mercadotecnia**

0.29 < 0.05

No rechazamos Ho.

La variable no tiene un efecto significativo, pertenecer al departamento de mercadoctenia no tiene efecto sobre la mediana del tiempo de supervivencia.

**departamentos__recursos humanos**

0.01 < 0.05 

Rechazamos Ho.

La variable tiene un efecto significativo, el pertenecer al departamento de recursos humanos disminuye la mediana del tiempo de supervivencia 8% con respecto a pertenecer al departamento de investigación.

**departamentos__soporte**

0.005 < 0.05

Rechazamos Ho. 

La variable tiene un efecto significativo, el pertenecer al departamento de soporte disminuye la mediana del tiempo de supervivencia 8% con respecto a pertenecer al departamento de investigación.

**departamentos__ventas**

0.03 < 0.05

Rechazamos Ho. 

La variable tiene un efecto significativo, el pertenecer al departamento de ventas disminuye la mediana del tiempo de supervivencia 6% con respecto a pertenecer al departamento de investigación.

**salarios__bajo**

0.005 < 0.05

Rechazamos Ho. 

La variable tiene efecto significativo, el ganar un salario bajo disminuye la mediana del tiempo de supervivencia 31% con respecto a recibir un salario alto.

**salarios__medio**

0.005 < 0.05

Rechazamos Ho. 

La variable tiene efecto significativo, el ganar un salario medio disminuye la mediana del tiempo de supervivencia 7% con respecto a recibir un salario alto.